src/WebJobs.Script.WebHost/Middleware/CorsMiddlewareFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.WebHost.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class CorsMiddlewareFactory : ICorsMiddlewareFactory
+    {
+        private IOptions<CorsOptions> _corsOptions;
+
+        public CorsMiddlewareFactory(IOptions<CorsOptions> corsOptions)
+        {
+            _corsOptions = corsOptions;
+        }
+
+        public CorsMiddleware CreateCorsMiddleware(RequestDelegate next, IOptions<HostCorsOptions> corsOptions)
+        {
+            CorsMiddleware middleware = null;
+            if (corsOptions?.Value != null)
+            {
+                var allowedOrigins = corsOptions.Value.AllowedOrigins?.ToArray();
+
+                var policyBuilder = allowedOrigins != null
+                    ? new CorsPolicyBuilder(allowedOrigins)
+                    : new CorsPolicyBuilder(new CorsPolicy());
+
+                if (corsOptions.Value.SupportCredentials)
```

Comment:
Could we add the `CorsOptions` configuration to a `CorsOptionsSetup` (implementing `IConfigureOptions<CorsOptions>` instead, keeping the configuration consistent?

Reply:
That sounds much cleaner; I'll implement it.